Apps/WebClient/src/NewClientApp/src/stores/patientData.ts

```diff
@@ -0,0 +1,272 @@
+                        PatientDataToHealthDataTypeMap.get(patientDataType)
+                )
+            );
+        }
+
+        patientDataMapUtil.setStateData(
+            patientDataMap.value,
+            hdid,
+            patientDataStateData
+        );
+    }
+
+    function areAllPatientDataTypesStored(
+        hdid: string,
+        patientDataTypes: PatientDataType[]
+    ): boolean {
+        const patientDataState = getPatientDataState(hdid);
+        if (!patientDataState.data) {
+            return false;
+        }
+        // Search for the first missing type. If none are missing, then its assumed all are stored. empty sets too.
+        return !patientDataTypes.some(
+            (patientDataType) => !patientDataState.data.has(patientDataType)
+        );
+    }
+
+    function handleError(
+        error: ResultError,
+        errorType: ErrorType,
+        patientDataTypes: PatientDataType[],
+        hdid: string,
+        fileId?: string
+    ): void {
+        logger.error(`ERROR: ${JSON.stringify(error)}`);
+        if (fileId) {
+            patientDataFileMapUtil.setStateError(
+                patientDataFilesMap.value,
+                fileId,
+                error
+            );
+        } else {
+            patientDataMapUtil.setStateError(patientDataMap.value, hdid, error);
+        }
+
+        if (error.statusCode === 429) {
+            errorStore.setTooManyRequestsWarning("page");
+        } else {
+            if (patientDataTypes && patientDataTypes.length > 0) {
+                patientDataTypes.forEach((patientDataType) => {
+                    errorStore.addError(
+                        errorType,
+                        getErrorSource(patientDataType),
+                        error.traceId
+                    );
+                });
+            } else {
+                errorStore.addError(
+                    errorType,
+                    ErrorSourceType.PatientData,
+                    error.traceId
+                );
+            }
+        }
+    }
+
+    function retrievePatientData(
+        hdid: string,
+        patientDataTypes: PatientDataType[]
+    ): Promise<PatientData[]> {
+        if (
+            getPatientDataState(hdid).status === LoadStatus.LOADED &&
+            areAllPatientDataTypesStored(hdid, patientDataTypes)
+        ) {
+            logger.debug("Patient data found stored, not querying!");
+            return Promise.resolve(patientData(hdid, patientDataTypes));
+        }
+        return patientDataService
+            .getPatientData(hdid, patientDataTypes)
+            .then((data) => {
+                setPatientData(hdid, patientDataTypes, data.items);
+                reportDataLoaded(patientDataTypes, data);
+                return data.items;
+            })
+            .catch((error: ResultError) => {
+                handleError(error, ErrorType.Retrieve, patientDataTypes, hdid);
+                return [];
+            });
```

Comment:
This would previously reject when unsuccessful.

Reply:
this was a copilot auto-complete miss. Thanks